src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/on_driver_distraction_notification_test.cc

```diff
@@ -122,40 +145,79 @@ TEST_F(HMIOnDriverDistractionNotificationTest,
+      .WillByDefault(Return(boost::optional<std::string>(kLSDW_message_en)));
+
   policy::CheckPermissionResult result;
   result.hmi_level_permitted = policy::kRpcAllowed;
   EXPECT_CALL(mock_policy_handler_interface_, CheckPermissions(_, _, _, _))
       .WillOnce(GetArg3(&result));
 
+  MessageSharedPtr message_to_mobile(
+      CreateMessage(smart_objects::SmartType_Map));
   EXPECT_CALL(mock_rpc_service_,
               ManageMobileCommand(
                   CheckNotificationParams(
                       am::mobile_api::FunctionID::OnDriverDistractionID, state),
-                  Command::CommandSource::SOURCE_SDL));
+                  Command::CommandSource::SOURCE_SDL))
+      .WillOnce(DoAll(SaveArg<0>(&message_to_mobile), Return(true)));
   command->Run();
+
+  ASSERT_TRUE((*message_to_mobile)[am::strings::msg_params].keyExists(
+      am::mobile_notification::lock_screen_dismissal_warning));
+
+  auto LSDW_message =
+      (*message_to_mobile)
+          [am::strings::msg_params]
+          [am::mobile_notification::lock_screen_dismissal_warning]
+              .asString();
+
+  EXPECT_EQ(kLSDW_message_en, LSDW_message);
 }
 
 TEST_F(HMIOnDriverDistractionNotificationTest,
        Run_PushMobileMessage_If_DisallowedByPolicy) {
   const auto state = hmi_apis::Common_DriverDistractionState::DD_ON;
   MessageSharedPtr commands_msg(CreateMessage(smart_objects::SmartType_Map));
   (*commands_msg)[am::strings::msg_params][am::hmi_notification::state] = state;
+  (*commands_msg)[am::strings::params][am::strings::connection_key] =
+      kConnectionKey;
   NotificationPtr command(
       CreateCommand<OnDriverDistractionNotification>(commands_msg));
 
   ON_CALL(mock_policy_handler_interface_, LockScreenDismissalEnabledState())
-      .WillByDefault(Return(OptionalBool(true)));
+      .WillByDefault(Return(boost::optional<bool>(true)));
+
+  std::string required_language = "en-us";
+  ON_CALL(mock_policy_handler_interface_,
+          LockScreenDismissalWarningMessage(required_language))
+      .WillByDefault(Return(boost::optional<std::string>(kLSDW_message_en)));
 
   policy::CheckPermissionResult result;
   result.hmi_level_permitted = policy::kRpcDisallowed;
   EXPECT_CALL(mock_policy_handler_interface_, CheckPermissions(_, _, _, _))
       .WillOnce(GetArg3(&result));
+
+  MessageSharedPtr pushed_message(CreateMessage(smart_objects::SmartType_Map));
   EXPECT_CALL(*mock_app_,
               PushMobileMessage(CheckNotificationParams(
-                  am::mobile_api::FunctionID::OnDriverDistractionID, state)));
+                  am::mobile_api::FunctionID::OnDriverDistractionID, state)))
+      .WillOnce(SaveArg<0>(&pushed_message));
 
   EXPECT_CALL(app_mngr_, set_driver_distraction_state(Eq(state)));
   command->Run();
+
+  ASSERT_TRUE((*pushed_message)[am::strings::msg_params].keyExists(
+      am::mobile_notification::lock_screen_dismissal_warning));
+
+  auto LSDW_message =
```

Comment:
@IGapchuk  I would also add UT with some not ascii charters

Reply:
@LuxoftAKutsan Fixed in 1453eb695bf77457789f9b9260b0a365926eaa39